io-engine/src/bdev/nexus/nexus_bdev_children.rs

```diff
@@ -739,66 +773,64 @@ fn update_failfast_done(
 }
 
 impl<'n> Nexus<'n> {
-    /// Marks a child device as faulted.
-    /// Returns true if the child was in open state, false otherwise.
-    fn child_io_faulted(
-        self: Pin<&mut Self>,
-        device_name: &str,
-        reason: Reason,
-    ) -> bool {
-        match self.lookup_child_device(device_name) {
-            Some(c) => {
-                debug!("{:?}: faulting with {}...", c, reason);
-
-                if Ok(ChildState::Open)
-                    == c.state.compare_exchange(
-                        ChildState::Open,
-                        ChildState::Faulted(reason),
-                    )
-                {
-                    warn!("{:?}: I/O faulted; will retire", c);
-                    true
-                } else {
-                    warn!("{:?}: I/O faulted; child was already faulted", c);
-                    false
-                }
-            }
-            None => {
-                error!(
-                    "{:?}: trying to fault a child device which \
-                        cannot be not found '{}'",
-                    self, device_name
-                );
-                false
-            }
-        }
-    }
-
-    /// TODO
+    /// Faults the device by its name, with the given fault reason.
+    /// The faulted device is scheduled to be retired.
     pub(crate) fn retire_child_device(
         mut self: Pin<&mut Self>,
         child_device: &str,
-        reason: Reason,
+        reason: FaultReason,
         retry: bool,
     ) {
-        // check if this child needs to be retired
-        let need_retire = self.as_mut().child_io_faulted(child_device, reason);
-
-        // The child state was not faulted yet, so this is the first I/O
-        // to this child for which we encountered an error.
-        if need_retire {
-            Reactors::master().send_future(Nexus::child_retire_routine(
-                self.name.clone(),
-                child_device.to_owned(),
-                retry,
-            ));
+        let name = self.name.clone();
+
+        if let Some(c) = self.as_mut().lookup_child_by_device_mut(child_device)
+        {
```

Comment:
just wondering, single a letter variable name is considered ok and RUST specific? Normally in C/C++, we force people to avoid single-letter vars since it makes it harder to maintain over some time

Reply:
That's matter of convention of a particular team or project. Oftentimes, even if a convention requires elaborated names, a single letter vars are okay for a scope of a "for" loop, like the index; or a similar kind of var for "if" scope like here.

But yes indeed, Rust conventions in general prefer shorter names than Java or C++. Rust does automatic type deduction in many cases when C/C++ does not. Thus it is impractical to encode type in var name like some C/C++ conventions do.

Also,  Rust IDEs are much more powerful than C/C++, and editors can be configured to show a small label showing data type for each var.